api/notd/sub_collection_token_manager.py

```diff
@@ -0,0 +1,22 @@
+from notd.store.retriever import Retriever
+from notd.store.saver import Saver
+from notd.sub_collection_token_processor import SubCollectionTokenProcessor
+from core.exceptions import NotFoundException
+
+class SubCollectionTokenManager:
+
+    def __init__(self, retriever: Retriever, saver: Saver, subCollectionTokenProcessor: SubCollectionTokenProcessor) -> None:
+        self.retriever = retriever
+        self.saver = saver
+        self.subCollectionTokenProcessor = subCollectionTokenProcessor
+
+    async def update_sub_collection_token(self,registryAddress: str, tokenId: str):
+        collectionName = await self.subCollectionTokenProcessor.retrieve_collection_name(registryAddress=registryAddress, tokenId=tokenId)
```

Comment:
this is where you deal with non-sub collections. if someone calls this with a normal collection (when retrieve_collection_name returns None) this should fail